File: Chronos.App/ClientApp/src/app/models/blank/blank-story.ts
import { User } from "src/app/service";
import { BlankTask } from "./blank-task";

export class BlankStory {
    storyId = 0;
    name = '';
    tasks: BlankTask[] = null;
    orderNumber = 0;
    assigned: User = null;
    isCollapsed = true;

    constructor(obj?: Partial<BlankStory>) {
        if (obj)
            Object.assign(this, obj);

        if (obj.tasks) {
            let tasks:BlankTask[] = []; 

            obj.tasks.forEach(task => {
                tasks.push(new BlankTask(task));
            });

            this.tasks = tasks;
        }
    }
}

File: Chronos.App/DataContracts/Blank/BlankStory.cs
using System.Collections.Generic;

namespace Chronos.App.DataContracts.Blank
{
    public class BlankStory
    {
        public int StoryId { get; set; }
        
        public string Name { get; set; }

        public List<BlankTask> Tasks { get; set; }
        
        public int OrderNumber { get; set; }
        
        public List<string> Tags { get; set; }
        
        public List<string> SuggestedTags { get; set; }
    }
}
